ambari-server/src/main/java/org/apache/ambari/server/agent/stomp/AmbariSubscriptionRegistry.java

```diff
@@ -250,35 +256,42 @@ public String toString() {
         CacheBuilder.newBuilder().maximumSize(cacheLimit).build();
 
     public LinkedMultiValueMap<String, String> getSubscriptions(String destination, Message<?> message) {
+      LinkedMultiValueMap<String, String> copiedSubscriptions = new LinkedMultiValueMap<>();
       if (notSubscriptionCache.asMap().keySet().contains(destination)) {
-        return new LinkedMultiValueMap<>();
+        return copiedSubscriptions;
       }
-      LinkedMultiValueMap<String, String> subscriptions = this.accessCache.computeIfAbsent(destination, (key) -> {
-        LinkedMultiValueMap<String, String> result = new LinkedMultiValueMap<>();
-        for (SessionSubscriptionInfo info : subscriptionRegistry.getAllSubscriptions()) {
-          for (String destinationPattern : info.getDestinations()) {
-            //TODO temporary changed to more fast-acting check without regex, need move investigation
-            if (destinationPattern.equals(destination)) {
-              for (Subscription subscription : info.getSubscriptions(destinationPattern)) {
-                result.add(info.sessionId, subscription.getId());
+      this.accessCache.compute(destination, (key, value) -> {
+        if (value == null) {
+          LinkedMultiValueMap<String, String> result = new LinkedMultiValueMap<>();
+          for (SessionSubscriptionInfo info : subscriptionRegistry.getAllSubscriptions()) {
```

Comment:
I see, tha in many places you starting to use actively streams, as reason there you could do the same - in spirit of similarity